items:
- uid: System.IO.FileNotFoundException
  commentId: T:System.IO.FileNotFoundException
  id: FileNotFoundException
  parent: System.IO
  children:
  - System.IO.FileNotFoundException.#ctor
  - System.IO.FileNotFoundException.#ctor(System.String)
  - System.IO.FileNotFoundException.#ctor(System.String,System.Exception)
  - System.IO.FileNotFoundException.#ctor(System.String,System.String)
  - System.IO.FileNotFoundException.#ctor(System.String,System.String,System.Exception)
  - System.IO.FileNotFoundException.FileName
  - System.IO.FileNotFoundException.Message
  - System.IO.FileNotFoundException.ToString
  langs:
  - csharp
  - vb
  name: FileNotFoundException
  fullName: System.IO.FileNotFoundException
  type: Class
  source:
    remote:
      path: src/mscorlib/src/System/IO/FileNotFoundException.cs
      branch: master
      repo: https://github.com/dotnet/coreclr.git
    id: FileNotFoundException
    path: src/mscorlib/src/System/IO/FileNotFoundException.cs
    startLine: 27
  assemblies:
  - System.Runtime
  namespace: System.IO
  summary: "\n<p>The exception that is thrown when an attempt to access a file that does not exist on disk fails.</p>\n"
  syntax:
    content: >-
      [ComVisible(true)]

      public class FileNotFoundException : IOException, ISerializable, _Exception
    content.vb: >-
      <ComVisible(True)>

      Public Class FileNotFoundException
          Inherits IOException
          Implements ISerializable, _Exception
  seealso:
  - type: System.Exception
    commentId: T:System.Exception
  inheritance:
  - System.Object
  - System.Exception
  - System.IO.IOException
  inheritedMembers:
  - System.Exception.Data
  - System.Exception.HelpLink
  - System.Exception.HResult
  - System.Exception.InnerException
  - System.Exception.Source
  - System.Exception.StackTrace
  - System.Exception.GetBaseException
  - System.Object.Equals(System.Object)
  - System.Object.Equals(System.Object,System.Object)
  - System.Object.ReferenceEquals(System.Object,System.Object)
  - System.Object.GetHashCode
  - System.Object.GetType
  - System.Object.MemberwiseClone
  modifiers.csharp:
  - public
  - class
  modifiers.vb:
  - Public
  - Class
  nameWithType: FileNotFoundException
- uid: System.IO.FileNotFoundException.#ctor
  commentId: M:System.IO.FileNotFoundException.#ctor
  id: '#ctor'
  parent: System.IO.FileNotFoundException
  langs:
  - csharp
  - vb
  name: FileNotFoundException()
  fullName: System.IO.FileNotFoundException.FileNotFoundException()
  type: Constructor
  source:
    remote:
      path: src/mscorlib/src/System/IO/FileNotFoundException.cs
      branch: master
      repo: https://github.com/dotnet/coreclr.git
    id: .ctor
    path: src/mscorlib/src/System/IO/FileNotFoundException.cs
    startLine: 33
  assemblies:
  - System.Runtime
  namespace: System.IO
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.IO.FileNotFoundException\" data-throw-if-not-resolved=\"false\"></xref> class with its message string set to a system-supplied message and its HRESULT set to COR_E_FILENOTFOUND.</p>\n"
  syntax:
    content: public FileNotFoundException()
    content.vb: Public Sub New
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
  nameWithType: FileNotFoundException.FileNotFoundException()
- uid: System.IO.FileNotFoundException.#ctor(System.String)
  commentId: M:System.IO.FileNotFoundException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.IO.FileNotFoundException
  langs:
  - csharp
  - vb
  name: FileNotFoundException(String)
  fullName: System.IO.FileNotFoundException.FileNotFoundException(System.String)
  type: Constructor
  source:
    remote:
      path: src/mscorlib/src/System/IO/FileNotFoundException.cs
      branch: master
      repo: https://github.com/dotnet/coreclr.git
    id: .ctor
    path: src/mscorlib/src/System/IO/FileNotFoundException.cs
    startLine: 38
  assemblies:
  - System.Runtime
  namespace: System.IO
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.IO.FileNotFoundException\" data-throw-if-not-resolved=\"false\"></xref> class with its message string set to <em>message</em> and its HRESULT set to COR_E_FILENOTFOUND.</p>\n"
  syntax:
    content: public FileNotFoundException(string message)
    content.vb: Public Sub New(message As String)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>A description of the error. The content of <em>message</em> is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture. </p>\n"
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
  nameWithType: FileNotFoundException.FileNotFoundException(String)
- uid: System.IO.FileNotFoundException.#ctor(System.String,System.Exception)
  commentId: M:System.IO.FileNotFoundException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.IO.FileNotFoundException
  langs:
  - csharp
  - vb
  name: FileNotFoundException(String, Exception)
  fullName: System.IO.FileNotFoundException.FileNotFoundException(System.String, System.Exception)
  type: Constructor
  source:
    remote:
      path: src/mscorlib/src/System/IO/FileNotFoundException.cs
      branch: master
      repo: https://github.com/dotnet/coreclr.git
    id: .ctor
    path: src/mscorlib/src/System/IO/FileNotFoundException.cs
    startLine: 43
  assemblies:
  - System.Runtime
  namespace: System.IO
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.IO.FileNotFoundException\" data-throw-if-not-resolved=\"false\"></xref> class with a specified error message and a reference to the inner exception that is the cause of this exception.</p>\n"
  syntax:
    content: public FileNotFoundException(string message, Exception innerException)
    content.vb: Public Sub New(message As String, innerException As Exception)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>A description of the error. The content of <em>message</em> is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture. </p>\n"
    - id: innerException
      type: System.Exception
      description: "\n<p>The exception that is the cause of the current exception. If the <em>innerException</em> parameter is not <strong>null</strong>, the current exception is raised in a <strong>catch</strong> block that handles the inner exception. </p>\n"
  seealso:
  - type: System.Exception
    commentId: T:System.Exception
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
  nameWithType: FileNotFoundException.FileNotFoundException(String, Exception)
- uid: System.IO.FileNotFoundException.#ctor(System.String,System.String)
  commentId: M:System.IO.FileNotFoundException.#ctor(System.String,System.String)
  id: '#ctor(System.String,System.String)'
  parent: System.IO.FileNotFoundException
  langs:
  - csharp
  - vb
  name: FileNotFoundException(String, String)
  fullName: System.IO.FileNotFoundException.FileNotFoundException(System.String, System.String)
  type: Constructor
  source:
    remote:
      path: src/mscorlib/src/System/IO/FileNotFoundException.cs
      branch: master
      repo: https://github.com/dotnet/coreclr.git
    id: .ctor
    path: src/mscorlib/src/System/IO/FileNotFoundException.cs
    startLine: 48
  assemblies:
  - System.Runtime
  namespace: System.IO
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.IO.FileNotFoundException\" data-throw-if-not-resolved=\"false\"></xref> class with its message string set to <em>message</em>, specifying the file name that cannot be found, and its HRESULT set to COR_E_FILENOTFOUND.</p>\n"
  syntax:
    content: public FileNotFoundException(string message, string fileName)
    content.vb: Public Sub New(message As String, fileName As String)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>A description of the error. The content of <em>message</em> is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture. </p>\n"
    - id: fileName
      type: System.String
      description: "\n<p>The full name of the file with the invalid image. </p>\n"
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
  nameWithType: FileNotFoundException.FileNotFoundException(String, String)
- uid: System.IO.FileNotFoundException.#ctor(System.String,System.String,System.Exception)
  commentId: M:System.IO.FileNotFoundException.#ctor(System.String,System.String,System.Exception)
  id: '#ctor(System.String,System.String,System.Exception)'
  parent: System.IO.FileNotFoundException
  langs:
  - csharp
  - vb
  name: FileNotFoundException(String, String, Exception)
  fullName: System.IO.FileNotFoundException.FileNotFoundException(System.String, System.String, System.Exception)
  type: Constructor
  source:
    remote:
      path: src/mscorlib/src/System/IO/FileNotFoundException.cs
      branch: master
      repo: https://github.com/dotnet/coreclr.git
    id: .ctor
    path: src/mscorlib/src/System/IO/FileNotFoundException.cs
    startLine: 54
  assemblies:
  - System.Runtime
  namespace: System.IO
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.IO.FileNotFoundException\" data-throw-if-not-resolved=\"false\"></xref> class with a specified error message and a reference to the inner exception that is the cause of this exception.</p>\n"
  syntax:
    content: public FileNotFoundException(string message, string fileName, Exception innerException)
    content.vb: Public Sub New(message As String, fileName As String, innerException As Exception)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>The error message that explains the reason for the exception. </p>\n"
    - id: fileName
      type: System.String
      description: "\n<p>The full name of the file with the invalid image. </p>\n"
    - id: innerException
      type: System.Exception
      description: "\n<p>The exception that is the cause of the current exception. If the <em>innerException</em> parameter is not <strong>null</strong>, the current exception is raised in a <strong>catch</strong> block that handles the inner exception. </p>\n"
  seealso:
  - type: System.Exception
    commentId: T:System.Exception
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
  nameWithType: FileNotFoundException.FileNotFoundException(String, String, Exception)
- uid: System.IO.FileNotFoundException.FileName
  commentId: P:System.IO.FileNotFoundException.FileName
  id: FileName
  parent: System.IO.FileNotFoundException
  langs:
  - csharp
  - vb
  name: FileName
  fullName: System.IO.FileNotFoundException.FileName
  type: Property
  source:
    remote:
      path: src/mscorlib/src/System/IO/FileNotFoundException.cs
      branch: master
      repo: https://github.com/dotnet/coreclr.git
    id: FileName
    path: src/mscorlib/src/System/IO/FileNotFoundException.cs
    startLine: 80
  assemblies:
  - System.Runtime
  namespace: System.IO
  summary: "\n<p>Gets the name of the file that cannot be found.</p>\n"
  syntax:
    content: public string FileName { get; }
    content.vb: Public ReadOnly Property FileName As String
    parameters: []
    return:
      type: System.String
      description: "\n<p>The name of the file, or <strong>null</strong> if no file name was passed to the constructor for this instance.</p>\n"
  modifiers.csharp:
  - public
  - get
  modifiers.vb:
  - Public
  - ReadOnly
  nameWithType: FileNotFoundException.FileName
- uid: System.IO.FileNotFoundException.Message
  commentId: P:System.IO.FileNotFoundException.Message
  id: Message
  parent: System.IO.FileNotFoundException
  langs:
  - csharp
  - vb
  name: Message
  fullName: System.IO.FileNotFoundException.Message
  type: Property
  source:
    remote:
      path: src/mscorlib/src/System/IO/FileNotFoundException.cs
      branch: master
      repo: https://github.com/dotnet/coreclr.git
    id: Message
    path: src/mscorlib/src/System/IO/FileNotFoundException.cs
    startLine: 60
  assemblies:
  - System.Runtime
  namespace: System.IO
  summary: "\n<p>Gets the error message that explains the reason for the exception.</p>\n"
  syntax:
    content: public override string Message { get; }
    content.vb: Public Overrides ReadOnly Property Message As String
    parameters: []
    return:
      type: System.String
      description: "\n<p>The error message.</p>\n"
  overridden: System.Exception.Message
  modifiers.csharp:
  - public
  - override
  - get
  modifiers.vb:
  - Public
  - Overrides
  - ReadOnly
  nameWithType: FileNotFoundException.Message
- uid: System.IO.FileNotFoundException.ToString
  commentId: M:System.IO.FileNotFoundException.ToString
  id: ToString
  parent: System.IO.FileNotFoundException
  langs:
  - csharp
  - vb
  name: ToString()
  fullName: System.IO.FileNotFoundException.ToString()
  type: Method
  source:
    remote:
      path: src/mscorlib/src/System/IO/FileNotFoundException.cs
      branch: master
      repo: https://github.com/dotnet/coreclr.git
    id: ToString
    path: src/mscorlib/src/System/IO/FileNotFoundException.cs
    startLine: 84
  assemblies:
  - System.Runtime
  namespace: System.IO
  summary: "\n<p>Returns the fully qualified name of this exception and possibly the error message, the name of the inner exception, and the stack trace.</p>\n"
  syntax:
    content: public override string ToString()
    content.vb: Public Overrides Function ToString As String
    return:
      type: System.String
      description: "\n<p>The fully qualified name of this exception and possibly the error message, the name of the inner exception, and the stack trace.</p>\n"
  overridden: System.Exception.ToString
  modifiers.csharp:
  - public
  - override
  modifiers.vb:
  - Public
  - Overrides
  nameWithType: FileNotFoundException.ToString()
references:
- uid: System.IO
  isExternal: false
  name: System.IO
  fullName: System.IO
  nameWithType: System.IO
  commentId: N:System.IO
- uid: System.Object
  parent: System
  isExternal: false
  name: Object
  fullName: System.Object
  nameWithType: Object
  commentId: T:System.Object
- uid: System.Exception
  parent: System
  isExternal: false
  name: Exception
  fullName: System.Exception
  nameWithType: Exception
  commentId: T:System.Exception
- uid: System.IO.IOException
  parent: System.IO
  isExternal: false
  name: IOException
  fullName: System.IO.IOException
  nameWithType: IOException
  commentId: T:System.IO.IOException
- uid: System.Exception.Data
  parent: System.Exception
  isExternal: false
  name: Data
  fullName: System.Exception.Data
  nameWithType: Exception.Data
  commentId: P:System.Exception.Data
- uid: System.Exception.HelpLink
  parent: System.Exception
  isExternal: false
  name: HelpLink
  fullName: System.Exception.HelpLink
  nameWithType: Exception.HelpLink
  commentId: P:System.Exception.HelpLink
- uid: System.Exception.HResult
  parent: System.Exception
  isExternal: false
  name: HResult
  fullName: System.Exception.HResult
  nameWithType: Exception.HResult
  commentId: P:System.Exception.HResult
- uid: System.Exception.InnerException
  parent: System.Exception
  isExternal: false
  name: InnerException
  fullName: System.Exception.InnerException
  nameWithType: Exception.InnerException
  commentId: P:System.Exception.InnerException
- uid: System.Exception.Source
  parent: System.Exception
  isExternal: false
  name: Source
  fullName: System.Exception.Source
  nameWithType: Exception.Source
  commentId: P:System.Exception.Source
- uid: System.Exception.StackTrace
  parent: System.Exception
  isExternal: false
  name: StackTrace
  fullName: System.Exception.StackTrace
  nameWithType: Exception.StackTrace
  commentId: P:System.Exception.StackTrace
- uid: System.Exception.GetBaseException
  parent: System.Exception
  isExternal: false
  name: GetBaseException()
  fullName: System.Exception.GetBaseException()
  spec.csharp:
  - uid: System.Exception.GetBaseException
    name: GetBaseException
    fullName: System.Exception.GetBaseException
  - name: (
    fullName: (
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Exception.GetBaseException
    name: GetBaseException
    fullName: System.Exception.GetBaseException
  - name: (
    fullName: (
  - name: )
    fullName: )
  nameWithType: Exception.GetBaseException()
  commentId: M:System.Exception.GetBaseException
- uid: System.Object.Equals(System.Object)
  parent: System.Object
  isExternal: true
  name: Equals(Object)
  fullName: System.Object.Equals(System.Object)
  spec.csharp:
  - uid: System.Object.Equals(System.Object)
    name: Equals
    fullName: System.Object.Equals
    isExternal: true
  - name: (
    fullName: (
  - uid: System.Object
    name: Object
    fullName: System.Object
    isExternal: true
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Object.Equals(System.Object)
    name: Equals
    fullName: System.Object.Equals
    isExternal: true
  - name: (
    fullName: (
  - uid: System.Object
    name: Object
    fullName: System.Object
    isExternal: true
  - name: )
    fullName: )
  nameWithType: Object.Equals(Object)
  commentId: M:System.Object.Equals(System.Object)
- uid: System.Object.Equals(System.Object,System.Object)
  parent: System.Object
  isExternal: true
  name: Equals(Object, Object)
  fullName: System.Object.Equals(System.Object, System.Object)
  spec.csharp:
  - uid: System.Object.Equals(System.Object,System.Object)
    name: Equals
    fullName: System.Object.Equals
    isExternal: true
  - name: (
    fullName: (
  - uid: System.Object
    name: Object
    fullName: System.Object
    isExternal: true
  - name: ', '
    fullName: ', '
  - uid: System.Object
    name: Object
    fullName: System.Object
    isExternal: true
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Object.Equals(System.Object,System.Object)
    name: Equals
    fullName: System.Object.Equals
    isExternal: true
  - name: (
    fullName: (
  - uid: System.Object
    name: Object
    fullName: System.Object
    isExternal: true
  - name: ', '
    fullName: ', '
  - uid: System.Object
    name: Object
    fullName: System.Object
    isExternal: true
  - name: )
    fullName: )
  nameWithType: Object.Equals(Object, Object)
  commentId: M:System.Object.Equals(System.Object,System.Object)
- uid: System.Object.ReferenceEquals(System.Object,System.Object)
  parent: System.Object
  isExternal: true
  name: ReferenceEquals(Object, Object)
  fullName: System.Object.ReferenceEquals(System.Object, System.Object)
  spec.csharp:
  - uid: System.Object.ReferenceEquals(System.Object,System.Object)
    name: ReferenceEquals
    fullName: System.Object.ReferenceEquals
    isExternal: true
  - name: (
    fullName: (
  - uid: System.Object
    name: Object
    fullName: System.Object
    isExternal: true
  - name: ', '
    fullName: ', '
  - uid: System.Object
    name: Object
    fullName: System.Object
    isExternal: true
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Object.ReferenceEquals(System.Object,System.Object)
    name: ReferenceEquals
    fullName: System.Object.ReferenceEquals
    isExternal: true
  - name: (
    fullName: (
  - uid: System.Object
    name: Object
    fullName: System.Object
    isExternal: true
  - name: ', '
    fullName: ', '
  - uid: System.Object
    name: Object
    fullName: System.Object
    isExternal: true
  - name: )
    fullName: )
  nameWithType: Object.ReferenceEquals(Object, Object)
  commentId: M:System.Object.ReferenceEquals(System.Object,System.Object)
- uid: System.Object.GetHashCode
  parent: System.Object
  isExternal: true
  name: GetHashCode()
  fullName: System.Object.GetHashCode()
  spec.csharp:
  - uid: System.Object.GetHashCode
    name: GetHashCode
    fullName: System.Object.GetHashCode
    isExternal: true
  - name: (
    fullName: (
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Object.GetHashCode
    name: GetHashCode
    fullName: System.Object.GetHashCode
    isExternal: true
  - name: (
    fullName: (
  - name: )
    fullName: )
  nameWithType: Object.GetHashCode()
  commentId: M:System.Object.GetHashCode
- uid: System.Object.GetType
  parent: System.Object
  isExternal: true
  name: GetType()
  fullName: System.Object.GetType()
  spec.csharp:
  - uid: System.Object.GetType
    name: GetType
    fullName: System.Object.GetType
    isExternal: true
  - name: (
    fullName: (
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Object.GetType
    name: GetType
    fullName: System.Object.GetType
    isExternal: true
  - name: (
    fullName: (
  - name: )
    fullName: )
  nameWithType: Object.GetType()
  commentId: M:System.Object.GetType
- uid: System.Object.MemberwiseClone
  parent: System.Object
  isExternal: true
  name: MemberwiseClone()
  fullName: System.Object.MemberwiseClone()
  spec.csharp:
  - uid: System.Object.MemberwiseClone
    name: MemberwiseClone
    fullName: System.Object.MemberwiseClone
    isExternal: true
  - name: (
    fullName: (
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Object.MemberwiseClone
    name: MemberwiseClone
    fullName: System.Object.MemberwiseClone
    isExternal: true
  - name: (
    fullName: (
  - name: )
    fullName: )
  nameWithType: Object.MemberwiseClone()
  commentId: M:System.Object.MemberwiseClone
- uid: System
  isExternal: false
  name: System
  fullName: System
  nameWithType: System
  commentId: N:System
- uid: System.String
  parent: System
  isExternal: false
  name: String
  fullName: System.String
  nameWithType: String
  commentId: T:System.String
- uid: System.Exception.Message
  parent: System.Exception
  isExternal: false
  name: Message
  fullName: System.Exception.Message
  nameWithType: Exception.Message
  commentId: P:System.Exception.Message
- uid: System.Exception.ToString
  parent: System.Exception
  isExternal: false
  name: ToString()
  fullName: System.Exception.ToString()
  spec.csharp:
  - uid: System.Exception.ToString
    name: ToString
    fullName: System.Exception.ToString
  - name: (
    fullName: (
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Exception.ToString
    name: ToString
    fullName: System.Exception.ToString
  - name: (
    fullName: (
  - name: )
    fullName: )
  nameWithType: Exception.ToString()
  commentId: M:System.Exception.ToString
